<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>crc</class>
 <widget class="QWidget" name="crc">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1274</width>
    <height>719</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Consolas</family>
    <pointsize>16</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>crc</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QLabel" name="label">
     <property name="font">
      <font>
       <family>楷体</family>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>CRC（循环冗余校验）计算工具</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="3,1">
     <item>
      <widget class="QLabel" name="label_2">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>需要校验的数据（十六进制，例如: AF1234 ）：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_clear">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>16</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>清空输入</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTextEdit" name="textEdit">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="documentTitle">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <family>楷体</family>
         <pointsize>16</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>参数模型 NAME:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>14</pointsize>
        </font>
       </property>
       <item>
        <property name="text">
         <string>自定义</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-4/ITU           x^4 + x + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-5/EPC           x^5 + x^3 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-5/ITU           x^5 + x^4 + x^2 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-5/USB           x^5 + x^2 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-6/ITU           x^6 + x + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-7/MMC           x^7 + x^3 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-8               x^8 + x^2 + x + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-8/ITU           x^8 + x^2 + x + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-8/ROHC          x^8 + x^2 + x + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-8/MAXIM         x^8 + x^5 + x^4 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-16/IBM          x^16 + x^15 + x^2 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-16/MAXIM        x^16 + x^15 + x^2 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-16/USB          x^16 + x^15 + x^2 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-16/MODBUS       x^16 + x^15 + x^2 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-16/CCITT        x^16 + x^12 + x^5 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-16/CCITT-FALSE  x^16 + x^12 + x^5 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-16/X25          x^16 + x^12 + x^5 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-16/XMODEM       x^16 + x^12 + x^5 + 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-16/DNP          x16+x13+x12+x11+x10+x8+x6+x5+x2+1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-32              x32+x26+x23+x22+x16+x12+x11+x10+x8+x7+x5+x4+x2+x+1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRC-32/MPEG-2       x32+x26+x23+x22+x16+x12+x11+x10+x8+x7+x5+x4+x2+x+1</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QFormLayout" name="formLayout">
       <property name="labelAlignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spinBox_width">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>32</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_5">
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>16</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>多项式 POLY(Hex):</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="lineEdit_poly"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_6">
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>16</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>初始值 INIT(Hex):</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="lineEdit_init"/>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_7">
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>16</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>结果异或值 XOROUT(Hex):</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="lineEdit_xorout"/>
       </item>
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="pushButton_table">
           <property name="font">
            <font>
             <family>楷体</family>
            </font>
           </property>
           <property name="text">
            <string>参数模型表</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_4">
           <property name="font">
            <font>
             <family>楷体</family>
             <pointsize>16</pointsize>
             <weight>50</weight>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>宽度 WIDTH：</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>例如：8</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>例如：ABCD</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>例如：FFFF</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>例如：00</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QCheckBox" name="checkBox_refin">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>输入数据反转（REFIN）</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkBox_refout">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>输出数据反转（REFOUT）</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="1,3">
     <item>
      <widget class="QPushButton" name="pushButton_calculate">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>70</width>
         <height>60</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>华文琥珀</family>
         <pointsize>24</pointsize>
        </font>
       </property>
       <property name="text">
        <string>计算</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_tips">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>16</pointsize>
         <weight>75</weight>
         <italic>true</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>请输入需要校验的数据 </string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QFormLayout" name="formLayout_2">
     <item row="0" column="0">
      <widget class="QLabel" name="label_13">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>校验结果（Hex）：</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="lineEdit_result_hex"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_14">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>校验结果（Bin）：</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="lineEdit_result_bin"/>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
